frontend/public/components/volumes-table.tsx

```diff
@@ -52,83 +44,107 @@ const ContainerLink: React.FC<ContainerLinkProps> = ({name, pod}) => <span class
-        <div className={classNames('col-lg-2 col-md-2 col-sm-3 hidden-xs co-break-all', { 'co-select-to-copy': _.get(m, 'subPath') })}>
-          {_.get(m, 'subPath', '-')}
-        </div>
-        <div className="col-lg-2 col-md-2 hidden-sm hidden-xs">
-          <VolumeIcon kind={kind} />
-          <span className="co-break-word">{loc && ` (${loc})`}</span>
-        </div>
-        <div className="col-lg-2 col-md-2 hidden-sm hidden-xs">{permission}</div>
-        <div className="col-lg-2 hidden-md hidden-sm hidden-xs">
-          { _.get(pod, 'kind') === 'Pod'
-            ? <ContainerLink name={m.container} pod={pod as PodKind} />
-            : <div>{m.container}</div>
-          }
-        </div>
-      </div>
-    </React.Fragment>)}
-  </div>;
+const volumeRoColumnClasses = [
+  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'col-xs-5'),
+  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'col-xs-7'),
+  classNames('col-lg-2', 'col-md-2', 'col-sm-4', 'hidden-xs'),
+  classNames('col-lg-2', 'col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'col-md-2', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'hidden-md', 'hidden-sm', 'hidden-xs'),
+  Kebab.columnClass,
+];
+
+const VolumesTableHeader = () => {
+  return [
+    {
+      title: 'Name', sortField: 'name', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[0]},
+    },
+    {
+      title: 'Mount Path', sortField: 'mountPath', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[1]},
+    },
+    {
+      title: 'SubPath', sortField: 'subPath', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[2]},
+    },
+    {
+      title: 'Type',
+      props: { className: volumeRoColumnClasses[3]},
+    },
+    {
+      title: 'Permissions', sortField: 'readOnly', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[4]},
+    },
+    {
+      title: 'Utilized By', sortField: 'container', transforms: [sortable],
+      props: { className: volumeRoColumnClasses[5]},
+    },
+    {
+      title: '',
+      props: { className: volumeRoColumnClasses[6]},
+    },
+  ];
 };
-VolumeRow.displayName = 'VolumeRow';
+VolumesTableHeader.displayName = 'VolumesTableHeader';
 
-export const VolumesTable: React.FC<VolumesTableProps> = ({podTemplate, heading}) => (
-  <React.Fragment>
-    {heading && <SectionHeading text={heading} />}
-    {_.isEmpty(podTemplate.spec.volumes)
+const VolumesTableRow = ({obj: volume, index, key, style}) => {
+  const type = _.get(getVolumeType(volume.volumeDetail), 'id', '');
+  const loc = getVolumeLocation(volume.volumeDetail);
+  const name = volume.name;
+  const permission = volume.readOnly ? 'Read-only' : 'Read/Write';
+  const { resource } = volume;
+  const pod = getResourceTemplate(resource);
+
+  return (
+    <TableRow id={index} index={index} trKey={key} style={style}>
```

Comment:
Is `id` an HTML `id` attribute? `index` does not seem like a good `id`. Is it required?

Reply:
It is required.  I changed it to name+mntPath for a unique row identifier.